packages/core/src/component.ts

```diff
@@ -3,4 +3,35 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-export interface Component {}
+import {Constructor, Provider} from '@loopback/context';
+import {Application} from '.';
+
+// tslint:disable:no-any
+
+export interface Component {
+  controllers?: Constructor<any>[];
+  providers?: {
+    [key: string]: Constructor<Provider<any>>;
+  };
+}
+
+export function mountComponent(
+  app: Application,
+  componentClass: Constructor<Component>,
+) {
+  const componentKey = 'components.' + componentClass.name;
+  app.bind(componentKey).toClass(componentClass);
+  const component = app.getSync(componentKey);
```

Comment:
Well if a component class cannot be instantiated synchronously, then we are in a big trouble. Right now, components are registered from Application constructor, which is always synchronous. If a component requires async computation to register, then how is the constructor going to wait for that computation to finish?